CHANGELOG.md

```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
```

Comment:
(Putting this here so you can answer)

@Scienfitz @AdrianSosic : Newest changes are implemented. I have also cleaned the git history, but did not do the force push yet. Only things blocking is the conversation regarding the use of `__file__` and that @AdrianSosic still needs to approve of the changes. So if you are happy with the current solution, please resolve/adjust review such that we can merge :)
If you prefer, I can also already force-push the cleaned git history such that you can have a more detailed look. This is what it looks like on my machine:
![image](https://github.com/emdgroup/baybe/assets/84016733/d9280036-293f-49b9-8e8c-af0ebaf71804)
Mainly summarized minor commits while making sure that commits that were not mine are being kept.